### PR TITLE
Add api-extractor

### PR DESCRIPTION
**Pull request by jeremymeng**
and format config json files

### CODE REVIEW

**Comment by jeremymeng** [comment]
Should be helpful in case we need to change any api later.

File: review/ms-rest-js.api.md [hunk 1]
```diff
@@ -0,0 +1,834 @@
...
+    addEventListener: (type: "abort", listener: (this: AbortSignalLike, ev: Event) => any, options?: any) => void;
+    // (undocumented)
+    dispatchEvent: (event: Event) => boolean;
+    // (undocumented)
+    onabort: ((this: AbortSignalLike, ev: Event) => any) | null;
+    // (undocumented)
+    removeEventListener: (type: "abort", listener: (this: AbortSignalLike, ev: Event) => any, options?: any) => void;
+}
+
+// Warning: (ae-forgotten-export) The symbol "AgentSettings" needs to be exported by the entry point msRest.d.ts
```
Comment (xirzec) [inline review/ms-rest-js.api.md:26]: missing export 👀

File: review/ms-rest-js.api.md [hunk 2]
```diff
@@ -0,0 +1,834 @@
...
+    };
+    HeaderConstants: {
+        AUTHORIZATION: string;
+        AUTHORIZATION_SCHEME: string;
+        RETRY_AFTER: string;
+        USER_AGENT: string;
+    };
+};
+
+// Warning: (ae-forgotten-export) The symbol "FetchHttpClient" needs to be exported by the entry point msRest.d.ts
```
Comment (xirzec) [inline review/ms-rest-js.api.md:172]: whoopsy, I think we ended up refactoring core-http because of this?

File: review/ms-rest-js.api.md [hunk 3]
```diff
@@ -0,0 +1,834 @@
...
+    // (undocumented)
+    prepareRequest(httpRequest: WebResourceLike): Promise<Partial<RequestInit>>;
+    // (undocumented)
+    processRequest(operationResponse: HttpOperationResponse): Promise<void>;
+}
+
+// @public
+export function delay<T>(t: number, value?: T): Promise<T>;
+
+// Warning: (ae-forgotten-export) The symbol "DeserializationContentTypes" needs to be exported by the entry point msRest.d.ts
```
Comment (xirzec) [inline review/ms-rest-js.api.md:191]: probably should export

File: review/ms-rest-js.api.md [hunk 4]
```diff
@@ -0,0 +1,834 @@
...
+// @public (undocumented)
+export function flattenResponse(_response: HttpOperationResponse, responseSpec: OperationResponse | undefined): RestResponse;
+
+// @public (undocumented)
+export function generateClientRequestIdPolicy(requestIdHeaderName?: string): RequestPolicyFactory;
+
+// @public
+export function generateUuid(): string;
+
+// Warning: (ae-forgotten-export) The symbol "ProxySettings" needs to be exported by the entry point msRest.d.ts
```
Comment (xirzec) [inline review/ms-rest-js.api.md:252]: export

File: review/ms-rest-js.api.md [hunk 5]
```diff
@@ -0,0 +1,834 @@
...
+// @public (undocumented)
+export function getDefaultUserAgentValue(): string;
+
+// @public
+export interface HttpClient extends RequestPolicy {
+}
+
+// @public
+export class HttpHeaders {
+    // Warning: (ae-forgotten-export) The symbol "RawHttpHeaders" needs to be exported by the entry point msRest.d.ts
```
Comment (xirzec) [inline review/ms-rest-js.api.md:266]: So many of these wow api-extractor bringing the value

File: package.json [hunk 6]
```diff
@@ -142,7 +143,8 @@
     "build:rollup": "rollup -c rollup.config.ts",
     "build:minify-browser": "terser -c -m --comments --source-map \"content='./dist/msRest.browser.js.map'\" -o ./dist/msRest.browser.min.js ./dist/msRest.browser.js",
     "build:test-browser": "webpack --config webpack.testconfig.ts",
-    "format": "prettier --write \"./**/*.ts\"",
+    "extract-api": "api-extractor run --local",
```
Comment (xirzec) [inline package.json:146]: should we also call this as part of `build`?

**Review by xirzec — APPROVED** [review]
Very good! Some minor comments. 

I think it may be worth fixing some of the missing exports when it is just a simple type (so basically everything except Fetch...)

File: package.json (hunk 6 above)
Comment (jeremymeng) [reply package.json:146]: added to `build:lib`

**Pull request merged**